src/main/ruby/core/posix.rb

```diff
@@ -152,6 +153,7 @@ def self.attach_function(native_name, argument_types, return_type,
   attach_function :umask, [:mode_t], :mode_t
   attach_function :unlink, [:string], :int
   attach_function :utimes, [:string, :pointer], :int
+  attach_function :write, [:int, :pointer, :size_t], :ssize_t
```

Comment:
You should declare this as `blocking: true`.

Reply:
Ah that is possible exactly what the problem is.